From jirislaby@gmail.com  Wed Mar 11 14:01:10 2009
From: Jiri Slaby <jirislaby@gmail.com>
Date: Wed, 11 Mar 2009 21:47:37 +0100
Subject: USB: image/mdc800, fix lock imbalance
To: gregkh@suse.de
Cc: Jiri Slaby <jirislaby@gmail.com>, Henning Zabel <henning@uni-paderborn.de>
Message-ID: <1236804460-1432-2-git-send-email-jirislaby@gmail.com>


There is an omitted unlock in mdc800_usb_probe's fail path. Add it.

Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
Cc: Henning Zabel <henning@uni-paderborn.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/image/mdc800.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/image/mdc800.c
+++ b/drivers/usb/image/mdc800.c
@@ -499,6 +499,7 @@ static int mdc800_usb_probe (struct usb_
 	retval = usb_register_dev(intf, &mdc800_class);
 	if (retval) {
 		dev_err(&intf->dev, "Not able to get a minor for this device.\n");
+		mutex_unlock(&mdc800->io_lock);
 		return -ENODEV;
 	}
 
